src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -90,7 +95,17 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
 
 
             var containerType = context.Key.ContainerType ?? context.Key.ModelType;
-            var localizer = _stringLocalizerFactory?.Create(containerType);
+            IStringLocalizer localizer;
+
+            if (_stringLocalizerFactory != null && _options?.Value?.DataAnnotationLocalizerProvider != null)
+            {
+                localizer = _options.Value.DataAnnotationLocalizerProvider(containerType, _stringLocalizerFactory);
+            }
+            else
+            {
+                localizer = _stringLocalizerFactory?.Create(containerType);
```

Comment:
Why does this fallback case exist? Not done in similar code elsewhere e.g. in the validator providers.